dev_guide/getting_traffic_into_cluster.adoc

```diff
@@ -0,0 +1,148 @@
+- If you have Encrypted TCP (i.e., TLS with the SNI header) -- Use the xref:../dev_guide/getting_traffic_into_cluster.adoc#using-a-router[Router],
+- If you have TCP or UDP -- Use xref:../dev_guide/getting_traffic_into_cluster.adoc#using-the-loadbalancer[LoadBalancer],
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-externalIP[ExternalIP], or
+xref:../dev_guide/getting_traffic_into_cluster.adoc#using-nodeport[NodePort].
+
+The router is the most common way to access the cluster.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+TCP or UDP is useful when the HTTP protocol is not being used or non-standard
+ports are in use. There is more manual setup and monitoring involved.
+
+The admin is responsible for configuring the external network so that
+the DNS name lands on the exposed IP. When setting up a router it can be
+convenient to use wildcard DNS since this lets the users self-provision
+the router.
+
+The exposed IP addresses are commonly the IP addresses of nodes, Virtual
+IP addresses (VRRP) or Service NodePort addreses, Service externalIPs,
+Service ingressIP, etc..
+
+The admin must make sure the local firewall on each node permits the
+request to reach the IP address.
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to make this highly available using replicated services.
+
+endif::[]
+
+
+[[using-a-router]]
+== Using a Router
+
+This is the most common way to access the cluster. A router is configured
+to accept external reuqests and proxy them based on the configured routes.
+This is limited to HTTP/HTTPS(SNI)/TLS(SNI) which covers web applications.
+
+An administrator can create a wild-card DNS entry, and then set up a router,
+after that the users can self-service the ingress without having to bother
+the admins. The router has controls to allow the admin to specify whether
+the users can self-provision hostnames, or if they must fit a pattern the
+admin defines. The other solutions require the admin to do it, or it requires
+that they delegate a lot of privilege.
+
+A set of routes can be created in the various projects. The overall
+set of routes is available to the set of routers. Each router selects
+from the set of routes.  All routers see all routes unless restricted
+by labels on the router, this is called router
+xref:../architecture/core_concepts/routes.adoc:#router-sharding[sharding].
+
+The xref:../admin_guide/high_availability.adoc#[High Availability] section
+describes how to configure a router for high availbility service using
+multiple replicas.
+
+[[using-the-loadbalancer]]
+== Using a Service of Type LoadBalancer
+
+There are load balancers available on
+xref:../install_config/configuring_aws.adoc#[AWS] and
+xref:../install_config/configuring_gce.adoc#[GEC] clouds as well as
+xref:../admin_guide/tcp_ingress_external_ports.adoc#[bare metal].
```

Comment:
Is it commonly understood that 'bare metal' can mean 'vm' too?  If not, it might make sense to say 'non-cloud' or something similar.  Assignment of ingress ip's can work on any cluster which does not assign load balancers to services of type LoadBalancer.